src/k8s/pkg/k8sd/types/cluster_config_convert.go

```diff
@@ -1,63 +1,55 @@
 package types
 
 import (
+	"fmt"
+	"strings"
+
 	apiv1 "github.com/canonical/k8s/api/v1"
 	"github.com/canonical/k8s/pkg/utils/vals"
 )
 
 // ClusterConfigFromBootstrapConfig converts BootstrapConfig from public API into a ClusterConfig.
-func ClusterConfigFromBootstrapConfig(b *apiv1.BootstrapConfig) ClusterConfig {
-	var config ClusterConfig
+func ClusterConfigFromBootstrapConfig(b apiv1.BootstrapConfig) (ClusterConfig, error) {
+	config := ClusterConfigFromUserFacing(b.ClusterConfig)
 
-	authorizationMode := "Node,RBAC"
-	if !vals.OptionalBool(b.EnableRBAC, true) {
-		authorizationMode = "AlwaysAllow"
+	// APIServer
+	config.APIServer.SecurePort = b.SecurePort
+	if b.DisableRBAC != nil && *b.DisableRBAC {
+		config.APIServer.AuthorizationMode = vals.Pointer("AlwaysAllow")
+	} else {
+		config.APIServer.AuthorizationMode = vals.Pointer("Node,RBAC")
 	}
-	config.APIServer.AuthorizationMode = vals.Pointer(authorizationMode)
 
-	switch b.Datastore {
+	// Datastore
+	switch b.GetDatastoreType() {
 	case "", "k8s-dqlite":
 		config.Datastore = Datastore{
 			Type:          vals.Pointer("k8s-dqlite"),
-			K8sDqlitePort: vals.Pointer(b.K8sDqlitePort),
+			K8sDqlitePort: b.K8sDqlitePort,
 		}
 	case "external":
+		if len(b.DatastoreServers) == 0 {
```

Comment:
Should we introduce a `BootstrapConfig.validate` method that checks that (and other validation, if any)?